site/docs/concepts/derivations.md

```diff
@@ -7,700 +7,769 @@ import Mermaid from '@theme/Mermaid';
 
 # Derivations
 
-A derivation is a [collection](../#collections)
-that results from transformations applied to one or more other collections.
+At times, the collections generated by a [capture](../#captures) may not be suitable for your needs.
+For instance, you might want to filter certain documents or add calculations to them.
+Perhaps you need to unpack an array nested inside or aggregate data from many documents.
+Alternatively, you might need to merge across several collections using a common key,
+or employ business logic to arrive at a real-time decision.
+With Flow derivations, you can perform a wide range of transformations,
+from a simple remapping to complicated, self-referential, and stateful transaction processing.
 
-Derivations derive data continuously,
-reflecting updates to the source collections as they happen.
+In essence, a derivation is a [collection](../#collections)
+that is constructed from applying transformations to one or more sourced collections.
+Derivations operate continuously, keeping up with updates to the source collections as they happen.
 
-A derivation has two main components:
+A derivation consists of three primary elements:
 
-* The new collection, which stores the derived data.
-* A [catalog task](../#tasks), which applies the transformations to source documents as they become available.
+* A collection that stores the output.
+* A [catalog task](../#tasks) that applies transformations to source documents
+  as they become available and writes the resulting documents into the derived collection.
+* An [internal task state](#internal-state) which enables aggregations, joins, and windowing.
 
-The derivation task is defined by:
+Today, Flow enables you to write derivations using either
+[SQLite](#sqlite) or [TypeScript](#typescript).
+Additional language support is in the works.
 
-* The **transformations** it applies
-* Its **register**, which serves as its internal memory
-* In many cases, **lambdas**, functions defined in accompanying TypeScript modules
-that allow more complex transformations.
+## Tutorial
 
-![](<derivations-new.svg>)
+### Introducing AcmeBank
 
-## Creating derivations
+The following tutorial sections will refer to the following common example
+to illustrate concepts.
```

Comment:
suggested reword (to set context and ensure readers don't over-agonize about understanding EVERYTHING from examples alone): 
The following tutorial sections use an illustrative example to introduce you to derivations, how you might use them, and their common components. We'll discuss each component in depth in subsequent sections of this page, but we recommend you start here to get your bearings.

Reply:
Done 👍